kotest-property/kotest-property-core/src/commonMain/kotlin/io/kotest/property/core/runtime.kt

```diff
@@ -0,0 +1,131 @@
+package io.kotest.property.core
+
+import io.kotest.assertions.print.print
+import io.kotest.common.ExperimentalKotest
+import io.kotest.core.names.TestName
+import io.kotest.core.spec.style.scopes.RootScope
+import io.kotest.core.spec.style.scopes.addTest
+import io.kotest.property.AssumptionFailedException
+import io.kotest.property.PropertyTesting
+import io.kotest.property.ShrinkingMode
+import io.kotest.property.internal.Counter
+import io.kotest.property.internal.ShrinkResult
+import io.kotest.property.internal.throwPropertyTestAssertionError
+
+@ExperimentalKotest
+fun RootScope.proptest(name: String, configure: PropTestContext.() -> Unit) {
+   addTest(TestName(name), false, null) {
+      val context = PropTestContext()
+      context.configure()
+      executePropTest(context)
+   }
+}
+
+internal suspend fun executePropTest(context: PropTestContext) {
+
+   if (context.seed != null && context.failOnSeed)
+      error("A seed is specified on this property test and failOnSeed is true")
+
+   val constraints = context.constraints
+      ?: context.iterations?.let { Constraints.iterations(it) }
+      ?: context.duration?.let { Constraints.duration(it) }
+      ?: Constraints.iterations(PropertyTesting.defaultIterationCount)
+
+   var k = 0
+   while (constraints.evaluate(Iteration(k))) {
+      try {
+         context.container.reset()
+         context.beforeProperty()
+         context.property.invoke()
+         context.afterProperty()
+      } catch (e: AssumptionFailedException) {
+         // we don't mark failed assumptions as errors
```

Comment:
Any chance we end up generating bad cases for the assumption forever?

It seems to me that if we pass by this, we might complete k iterations without a single test execution

Reply:
Yes, this is just PoC code though.